schemas/src/main/java/com/scottlogic/deg/schemas/v3/RuleDeserializer.java

```diff
@@ -13,25 +15,20 @@
     public RuleDTO deserialize(
         JsonParser jsonParser,
         DeserializationContext deserializationContext)
-        throws IOException, JsonProcessingException {
+        throws IOException {
 
         ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
 
         JsonNode node = mapper.readTree(jsonParser);
 
-        if (node.has("rule")) {
-            RuleDTO rule = new RuleDTO();
-            rule.rule = node.get("rule").asText();
-            rule.constraints = readConstraintsFromArrayNode(node.get("constraints"), mapper);
-            return rule;
-        }
-        else {
-            ConstraintDTO constraint = mapper.treeToValue(node, ConstraintDTO.class);
-
-            RuleDTO rule = new RuleDTO();
-            rule.constraints = Collections.singleton(constraint);
-            return rule;
-        }
+        RuleDTO rule = new RuleDTO();
```

Comment:
We need to unit test the change in logic here. Also have we considered the impact of the new return types? This will now return null for `rule` and an empty set for `constraints` if given invalid profiles - is this desired behaviour and how will it affect the consumers of this class?